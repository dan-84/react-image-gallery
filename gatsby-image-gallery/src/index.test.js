import React from 'react'
import Gallery from './index'
import ShallowRenderer from 'react-test-renderer/shallow'

const fluidShapeMock = (path) => ({
  aspectRatio: 1.5,
  src: path,
  srcSet: `some srcSet`,
  srcSetWebp: `some srcSetWebp`,
  sizes: `(max-width: 600px) 100vw, 600px`,
  base64: `string_of_base64`,
})

const unifiedImageShapeMock = (path) => ({
  thumb: fluidShapeMock(`/thumb${path}`),
  full: fluidShapeMock(path),
})

describe('Gallery component', () => {
  test('that it renders with empty props', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<Gallery images={[]} thumbs={[]} />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })

  test('that it renders with data in images', () => {
    const renderer = new ShallowRenderer()
    renderer.render(
      <Gallery
        images={[
          '/images/image001.jpg',
          '/images/image002.jpg',
          '/images/image003.jpg',
          '/images/image004.jpg',
        ]}
        thumbs={[]}
      />
    )
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })

  test('that it renders with data in thumbs', () => {
    const renderer = new ShallowRenderer()
    renderer.render(
      <Gallery
        images={[]}
        thumbs={[
          { fluid: fluidShapeMock('/thumb/image001.jpg') },
          { fluid: fluidShapeMock('/thumb/image002.jpg') },
          { fluid: fluidShapeMock('/thumb/image003.jpg') },
          { fluid: fluidShapeMock('/thumb/image004.jpg') },
          { fluid: fluidShapeMock('/thumb/image005.jpg') },
        ]}
      />
    )
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })

  test('that it renders with data in images & thumbs', () => {
    const renderer = new ShallowRenderer()
    renderer.render(
      <Gallery
        images={[
          '/images/image001.jpg',
          '/images/image002.jpg',
          '/images/image003.jpg',
          '/images/image004.jpg',
        ]}
        thumbs={[
          { fluid: fluidShapeMock('/thumb/image001.jpg') },
          { fluid: fluidShapeMock('/thumb/image002.jpg') },
          { fluid: fluidShapeMock('/thumb/image003.jpg') },
          { fluid: fluidShapeMock('/thumb/image004.jpg') },
          { fluid: fluidShapeMock('/thumb/image005.jpg') },
        ]}
      />
    )
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })

  test('that it renders with data in fullImages & thumbs', () => {
    const renderer = new ShallowRenderer()
    renderer.render(
      <Gallery
        fullImages={[
          '/images/image001.jpg',
          '/images/image002.jpg',
          '/images/image003.jpg',
          '/images/image004.jpg',
        ]}
        thumbs={[
          { fluid: fluidShapeMock('/thumb/image001.jpg') },
          { fluid: fluidShapeMock('/thumb/image002.jpg') },
          { fluid: fluidShapeMock('/thumb/image003.jpg') },
          { fluid: fluidShapeMock('/thumb/image004.jpg') },
          { fluid: fluidShapeMock('/thumb/image005.jpg') },
        ]}
      />
    )
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })

  test('that it renders with unified image prop', () => {
    const renderer = new ShallowRenderer()
    renderer.render(
      <Gallery
        images={[
          unifiedImageShapeMock('/images/image001.jpg'),
          unifiedImageShapeMock('/images/image002.jpg'),
          unifiedImageShapeMock('/images/image003.jpg'),
          unifiedImageShapeMock('/images/image004.jpg'),
          unifiedImageShapeMock('/images/image005.jpg'),
        ]}
      />
    )
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })

  test('that it renders with a custom column class', () => {
    const renderer = new ShallowRenderer()
    renderer.render(
      <Gallery
        colClass="col-md-3"
        images={[
          '/images/image001.jpg',
          '/images/image002.jpg',
          '/images/image003.jpg',
          '/images/image004.jpg',
        ]}
        thumbs={[
          { fluid: fluidShapeMock('/thumb/image001.jpg') },
          { fluid: fluidShapeMock('/thumb/image002.jpg') },
          { fluid: fluidShapeMock('/thumb/image003.jpg') },
          { fluid: fluidShapeMock('/thumb/image004.jpg') },
          { fluid: fluidShapeMock('/thumb/image005.jpg') },
        ]}
      />
    )
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
