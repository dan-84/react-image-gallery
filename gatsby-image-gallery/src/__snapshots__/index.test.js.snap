// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component that it renders with a custom column class 1`] = `
<React.Fragment>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image001.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image002.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image003.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image004.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image005.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
  </ForwardRef(styled.div)>
</React.Fragment>
`;

exports[`Gallery component that it renders with data in images & thumbs 1`] = `
<React.Fragment>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image001.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image002.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image003.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image004.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image005.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
  </ForwardRef(styled.div)>
</React.Fragment>
`;

exports[`Gallery component that it renders with data in images 1`] = `
<React.Fragment>
  <ForwardRef(styled.div) />
</React.Fragment>
`;

exports[`Gallery component that it renders with data in thumbs 1`] = `
<React.Fragment>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image001.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image002.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image003.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image004.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      md={25}
      onClick={[Function]}
      width={33.333333333333336}
    >
      <ForwardRef(styled.div)
        margin="0.25rem"
      >
        <Image
          Tag="div"
          alt=""
          className=""
          durationFadeIn={500}
          fadeIn={true}
          fluid={
            Object {
              "fluid": Object {
                "aspectRatio": 1.5,
                "base64": "string_of_base64",
                "sizes": "(max-width: 600px) 100vw, 600px",
                "src": "/thumb/image005.jpg",
                "srcSet": "some srcSet",
                "srcSetWebp": "some srcSetWebp",
              },
            }
          }
          loading="lazy"
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
  </ForwardRef(styled.div)>
</React.Fragment>
`;

exports[`Gallery component that it renders with empty props 1`] = `
<React.Fragment>
  <ForwardRef(styled.div) />
</React.Fragment>
`;
